### PR TITLE
UI-430 Grid component refactor

### PR DESCRIPTION
**Pull request by joshsylvester**
* Renamed components:
  * `Grid` => `Grid`
  * `Row` => `GridRow`
  * `Col` => `GridItem`
* Updated to match coding standards
* Broke out into multiple files
* Added tests & `propType` comments

### CODE REVIEW

File: lib/components/Grid/GridItem.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+const VALID_ALIGNS = ['top', 'medium', 'bottom'];
```
Comment (sheela-svmx) [inline lib/components/Grid/GridItem.js]: i am just curious what a medium align is? I know its their code , but its funny to align something as medium :)

Comment (hanzge) [reply lib/components/Grid/GridItem.js]: In css `vertical-align`, the values are `top`, `middle`, `bottom`, etc. `medium` looks like a size unit. If it's possible, I suggest to change to `middle`. However, if it's already in production usage, just let it be.

Comment (guohonghanusa) [reply lib/components/Grid/GridItem.js]: for vertical-align: top, middle, bottom. 
for horizontal-align: left, center, right.
The 'medium' should be 'middle'.

Comment (joshsylvester) [reply lib/components/Grid/GridItem.js]: I'll have to look into this - i assume it was a bad copy-paste from the original engineer (not from me)

Comment (joshsylvester) [reply lib/components/Grid/GridItem.js]: it turns out it was wrong from the orig developer, thanks for the catch! - fixed.

**Comment by mallik-svmx** [comment]
Two approvals, merging.

**Pull request merged**